common/predictive-text/worker-interface.d.ts

```diff
@@ -0,0 +1,41 @@
+// Necessary for headless compilation.  All contents were copied from the definitions in lib.dom.d.ts.
+
+/** A message received by a target object. */
+interface MessageEvent extends Event {
+  /**
+   * Returns the data of the message.
+   */
+  readonly data: any;
+  /**
+   * Returns the last event ID string, for server-sent events.
+   */
+  readonly lastEventId: string;
+  /**
+   * Returns the origin of the message, for server-sent events and cross-document messaging.
+   */
+  readonly origin: string;
+  /**
+   * Returns the MessagePort array sent with the message, for cross-document messaging and channel messaging.
+   */
+  readonly ports: ReadonlyArray<MessagePort>;
+  /**
+   * Returns the WindowProxy of the source window, for cross-document messaging, and the MessagePort being attached, in the connect event fired at SharedWorkerGlobalScope objects.
+   */
+  readonly source: MessageEventSource | null;
```

Comment:
Which properties of the `MessageEvent` does the LMLayer actually use? I 'm not sure it requires anything other than `data`. Since TypeScript supports [structural subtyping](https://www.typescriptlang.org/docs/handbook/type-compatibility.html#introduction), I would only keep the properties and methods that the LMLayer actually _uses_ in the type definition. This makes creating mocks and test much easier, because of the smaller footprint to mock.